### DISCUSSION TITLE
What's the recommended way to unit/integration test interaction?

### DISCUSSION THREAD

**Discussion by beeb** [fadeevab/cliclack#82] [Q&A]
Thanks for providing this great library. I was curious about how I can test CLI commands which use cliclack for interaction.

I couldn't find tests in this repo for inspiration so I have a hard time figuring out how I can do it.

Thanks!

**Comment by fadeevab** [comment]
@beeb Thanks! Well, this is a good question. I saw someone created proxy types for cliclack types, exposing `interact` method, and for the tests the `interact` method returned predefined values.

Perhaps, we have to bring new interact methods for testing to simplify testing. E.g. `interact_on` and/or `interact_mock`.

**Reply by beeb** [reply]
Having a way to mock interaction would be great indeed! Not sure how that would look in practice but maybe [mockall](https://crates.io/crates/mockall) could be used to mock the `PromptInteraction` trait. I'm no so familiar with Rust mocking so I can't go into specifics ^^
